js/form.js

```diff
@@ -22,6 +22,36 @@ window.addEventListener('load', function(){
     }
   });
 
+  /**
+   * Form Processing
+   * Disables all form fields and replaces the submit button with a span element whcih
+   * can be configured in the css
+   */
+  document.querySelectorAll('.fw-form').forEach(function(formElement){
```

Comment:
Could we update this to behave like the shoppingcart version? Then we can update/remove that from the shoppingcart.
- Use the same size as the source submit button
- Use the same position as the source submit button
- Remove "Processing..." if button is not wide

I tinkered with this a bit, and I think it would look something like:

```
  document.querySelectorAll('.fw-form').forEach(function(formElement){
    formElement.addEventListener('submit', function(event){

      // Add "Processing" element to replace submit button
      const processingElement = document.createElement("span");
      processingElement.className = "fw-form-processing";

      // Set processingElement to the appropriate size
      var submitterWidth = event.submitter.getBoundingClientRect().width;
      if(submitterWidth < 100){
        processingElement.classList.add('fw-form-processing-small');
      }
      processingElement.style.width = submitterWidth + 'px';
      processingElement.style.height = event.submitter.getBoundingClientRect().height + 'px';

      // Set processingElement to the appropriate spacing
      processingElement.style['margin-top'] = window.getComputedStyle(event.submitter)['margin-top'];
      processingElement.style['margin-right'] = window.getComputedStyle(event.submitter)['margin-right'];
      processingElement.style['margin-bottom'] = window.getComputedStyle(event.submitter)['margin-bottom'];
      processingElement.style['margin-left'] = window.getComputedStyle(event.submitter)['margin-left'];

      // Disable submit button, hide the button
      event.submitter.disabled = true;
      event.submitter.style.display = "none";

      // Position processingElement where submit button was
      event.submitter.parentElement.appendChild(processingElement);

      // Disable all fields
      var fields = this.getElementsByTagName('*');
      for(var i = 0; i < fields.length; i++)
      {
        fields[i].disabled = true;
        // Check for number input type
        if(fields[i].type=="number"){
          fields[i].parentElement.parentElement.classList.add("readonly");
        }
      }

    });
  });
```

The sass would need to include something like the following as well

```
  &.fw-form-processing-small{
    &:after{
      content: "";
      margin-left: 0;
    }
  }
```